### PR TITLE
docs: Add Cargo.toml style guide

### LINKED ISSUE #528: Document Cargo.toml formatting
It is good to have consistent formatting for consistency between developers (eg: using rustfmt for ust code) to reduce noise around style in PR reviews.

Automatic formatting is a preference, however exising TOML formatters (as far as we know) apply rules across the whole toml file (and not only Cargo.toml), rather than different rules per section or use case of toml.

For example,
- we would like `[dependencies]` to be ordered alphabetically
- we would like important `[project]` attributes to be at the top of the section

So, for now we should document how we expect Cargo.toml to look, to avoid it repeatedly coming up in PRs.

### PR DESCRIPTION
**Pull request by fhennig**
fixes #528

### CODE REVIEW

File: modules/contributor/pages/code-style-guide.adoc [hunk 1]
```diff
@@ -1,5 +1,55 @@
 = Source code style guide
 
+== Cargo.toml
+
+Follow the https://doc.rust-lang.org/nightly/style-guide/cargo.html[official formatting conventions] for the `Cargo.toml` file.
```
Comment (Techassi) [inline modules/contributor/pages/code-style-guide.adoc:5]: suggestion: Use we.

```suggestion
We follow the https://doc.rust-lang.org/nightly/style-guide/cargo.html[official formatting conventions] for the `Cargo.toml` file.
```

File: modules/contributor/pages/code-style-guide.adoc [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+This means:
+
+* Put the `[package]` section at the top of the file
+* put the name and version keys in that order at the top of that section, followed by the remaining keys other than description in order, followed by the description at the end of that section.
```
Comment (Techassi) [inline modules/contributor/pages/code-style-guide.adoc:9]: note: Capitalize the letter `p`.

Comment (Techassi) [inline modules/contributor/pages/code-style-guide.adoc:9]: note: What order are we talking about here? The same order as mentioned below? If so, we should link to it here already.

**Review by Techassi — CHANGES_REQUESTED** [review]

File: modules/contributor/pages/code-style-guide.adoc (hunk 1 above)
Comment (fhennig) [reply modules/contributor/pages/code-style-guide.adoc:5]: Actually, we shouldn't use "we" in our docs: https://docs.stackable.tech/home/stable/contributor/style-guide#_tone_and_writing_style_google_developer_documentation_style_guide

This is instructional "Do it like this"

Comment (Techassi) [reply modules/contributor/pages/code-style-guide.adoc:5]: Alright, then we should probably change that for the rest of the guide as well.

File: modules/contributor/pages/code-style-guide.adoc [hunk 3]
```diff
@@ -1,5 +1,55 @@
 = Source code style guide
 
+== Cargo.toml
+
+Follow the https://doc.rust-lang.org/nightly/style-guide/cargo.html[official formatting conventions] for the `Cargo.toml` file.
+This means:
+
+* Put the `[package]` section at the top of the file.
+* Put the name and version keys in that order at the top of that section, followed by the remaining keys other than description in order (sort keys with https://www.gnu.org/software/coreutils/manual/html_node/Version-sort-overview.html[version-sort]; very similar to lexical sorting)), followed by the description at the end of that section.
```
Comment (Techassi) [inline modules/contributor/pages/code-style-guide.adoc:9]: suggestion: Use inline code blocks.

```suggestion
* Put the `name` and `version` keys in that order at the top of that section, followed by the remaining keys other than `description` in order (sort keys with https://www.gnu.org/software/coreutils/manual/html_node/Version-sort-overview.html[version-sort]; very similar to lexical sorting)), followed by the `description` at the end of that section.
```

File: modules/contributor/pages/code-style-guide.adoc (hunk 1 above)
Comment (fhennig) [reply modules/contributor/pages/code-style-guide.adoc:5]: probably, but maybe not in this PR

Comment (fhennig) [reply modules/contributor/pages/code-style-guide.adoc:5]: or maybe I can do it :grin:

Comment (fhennig) [reply modules/contributor/pages/code-style-guide.adoc:5]: done

**Review by Techassi — APPROVED** [review]

**Pull request merged**